### ISSUE TITLE
whether related beta apis enabled for those  beta feature gates enabled by k8s 1.24 by default?

### ISSUE THREAD

**Issue by rxwang662001** [kubernetes/kubernetes#113713]
From https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
the statement

``Since Kubernetes 1.24, new beta APIs are not enabled by default. When enabling a beta feature, you will also need to enable any associated API resources. For example, to enable a particular resource like storage.k8s.io/v1beta1/csistoragecapacities, set --runtime-config=storage.k8s.io/v1beta1/csistoragecapacities. See [API Versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning) for more details on the command line flags.``

Since most of beta feature gates are enabled by default by k8s1.24 like below:
AnyVolumeDataSource true Beta 1.24
CSIMigrationAzureFile true Beta 1.24
GracefulNodeShutdownBasedOnPodPriority true Beta 1.24
GRPCContainerProbe true Beta 1.24
....

My concern is if these feature gates related any new beta APIs are also enabled by k8s1.24 by default? So user won't need to consider how to make these feature gate work completely on the cluster.

Thanks.

**Comment by pacoxu** [comment]
For `new beta APIs are not enabled by default`, it is for the KEP below with https://github.com/kubernetes/kubernetes/pull/107993.
https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/3136-beta-apis-off-by-default/README.md

This KEP is about beta API, and it is not-related to the feature gate beta concept.

> ### Non-Goals
> Change feature gate defaults. Feature gates control new features (not just new APIs) and they are on by default for beta features. This KEP is not changing the lifecycle flow for feature gates. It is currently alpha=off-by-default, beta=on-by-default, stable=locked-to-on.

For feature gate progressions, please refer to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md. beta API and beta feature gates most likely are different things.

**Comment by rxwang662001** [comment]
@pacoxu 
Yes, I also understand "beta feature gate" and "beta apis" are two things, that's why I have such confusion.
Since most of "beta feature gates" are enabled by default by k8s 1.24 still, and if their implementation have new beta APIs, whether these new beta APIs will be enabled by k8s 1.24 as well? so that these beta feature gates could work.
Or it still need user self to figure out those beta feature gates whether it have new beta APIs or not and enable them by user if wanted? and if then, what's a convenient way for user to figure these new beta apis out for each beta feature gates?

Thanks.

**Comment by pacoxu** [comment]
I think this should be mentioned in the feature gate document.

/cc @deads2k

**Comment by niranjandarshann** [comment]
Your concern is valid. Since Kubernetes 1.24, new beta APIs are not enabled by default, even if the corresponding feature gates are enabled. This means that enabling a beta feature gate does not automatically enable its associated API resources.

**Comment by seans3** [comment]
/triage accepted

Question: Should there be one atomic change: get feature and API together?

/sig architecture

**Comment by lmktfy** [comment]
I think this is answered. You do need to opt in to beta API groups even when feature gates are enabled by default.

There is no work to do.

/remove-triage accepted
/close not-planned
